gh-126204: Add a `sys._jit_enabled` helper (#126247)
python/cpython: Doc/whatsnew/3.14.rst

  which only exists in specialized builds of Python, may now return objects
  from other interpreters than the one it's called in.

> * Add a :func:`sys._jit_enabled` which helps to determine in runtime if
>   :ref:`JIT compiler <whatsnew313-jit-compiler>` is active.

Reviewer: ```suggestion
* Add :func:`sys._jit_enabled` to determine at runtime if the
  :ref:`JIT compiler <whatsnew313-jit-compiler>` is active.
```
